expected/16.0/extra/nogis.out

```diff
@@ -0,0 +1,158 @@
+ERROR:  This PostGIS data type is supported by SpatiaLite, but FDW compiled without GIS data support
+HINT:  SQLite value with "blob" affinity (60 bytes) in hex : 0001e6100000bf72ce99fe763e40ed4960730ed84d40bf72ce99fe763e40ed4960730ed84d407c01000000bf72ce99fe763e40ed4960730ed84d40fe
+CONTEXT:  foreign table "types_PostGIS" foreign column "gm" have data type "geometry" (usual affinity "blob"), in query there is reference to foreign column
+--Testcase 38:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT "i", gm, gg, t FROM "types_PostGIS";
+                              QUERY PLAN                               
+-----------------------------------------------------------------------
+ Foreign Scan on public."types_PostGIS"
+   Output: i, gm, gg, t
+   SQLite query: SELECT `i`, `gm`, `gg`, `t` FROM main."types_PostGIS"
+(3 rows)
+
+--Testcase 39:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT "i", gm, gg, t FROM "types_PostGIS" WHERE gm = '0101000020e6100000bf72ce99fe763e40ed4960730ed84d40'::geometry;
+                                                                        QUERY PLAN                                                                         
+-----------------------------------------------------------------------------------------------------------------------------------------------------------
+ Foreign Scan on public."types_PostGIS"
+   Output: i, gm, gg, t
+   Filter: (("types_PostGIS".gm)::bytea = '\x3031303130303030323065363130303030306266373263653939666537363365343065643439363037333065643834643430'::bytea)
+   SQLite query: SELECT `i`, `gm`, `gg`, `t` FROM main."types_PostGIS"
+(4 rows)
+
+--Testcase 40: ERR - no GIS data support
+SELECT "i", gm, gg, t FROM "types_PostGIS" WHERE gm = '0101000020e6100000bf72ce99fe763e40ed4960730ed84d40'::geometry;
+ERROR:  This PostGIS data type is supported by SpatiaLite, but FDW compiled without GIS data support
+HINT:  SQLite value with "blob" affinity (60 bytes) in hex : 0001e6100000bf72ce99fe763e40ed4960730ed84d40bf72ce99fe763e40ed4960730ed84d407c01000000bf72ce99fe763e40ed4960730ed84d40fe
+CONTEXT:  foreign table "types_PostGIS" foreign column "gm" have data type "geometry" (usual affinity "blob"), in query there is reference to foreign column
+-- Insert PostGIS/GEOS BLOB, read SpatiaLite BLOB
+--Testcase 41: ERR - no GIS data support
+INSERT INTO "types_PostGIS" ( "i", gm, gg, t ) VALUES (2, decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex'),  decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex'), '{"genus": "Rhododendron", "taxon": "Rhododendron ledebourii"}');
+ERROR:  This PostGIS data type is supported by SpatiaLite, but FDW compiled without GIS data support
+HINT:  Data type: "public.geometry" in column "gm"
+--Testcase 42:
+ALTER FOREIGN TABLE "types_PostGIS" ALTER COLUMN "gm" TYPE bytea;
+--Testcase 43:
+ALTER FOREIGN TABLE "types_PostGIS" ALTER COLUMN "gg" TYPE bytea;
+--Testcase 44: OK
+SELECT "i", gm, gg, t FROM "types_PostGIS";
```

Comment:
When domain `geometry` and `geography` are declared as domain of `bytea`, but they give error when select.  
However, changing them directly to bytea can give correct result.  
It is strange, because in both cases, basically, the actual data type is still `bytea`.
Is there any way to select data correct without error, even though we declare them as domain of `bytea`?

Reply:
> When domain `geometry` and `geography` are declared as domain of `bytea`, but they give error when select.

Yes, this is normal. Declaring as domains is not enough for GIS data transformation. User can have no PostGIS, but user cannot compile the FDW without GIS support for such data transformation.

Reply:
@mkgrgis I don't think this is a normal behavior. In this case, domain is just like an alias for `bytea`, so current behavior is inconsistent when declaring or not declaring domain.  
For domain type, please do not raise error immediately. We should get the base type of domain. If it is `bytea`, it can be transformed. If it is not `bytea`, raise error.  
I tried to modify sqlite_query.c.
`sqlite_convert_to_pg` function (for SELECT query):
```
		Oid			atttypid = att->atttypid;
		int32		atttypmod = att->atttypmod;

		/*
		 * If a domain has been declared as bytea, it can support PostGIS data type
		 */
		atttypid = getBaseTypeAndTypmod(atttypid, &atttypmod);

		if (atttypid == BYTEAOID)
		{
			value_datum = (Datum) palloc0(value_byte_size_blob_or_utf8 + VARHDRSZ);
			memcpy(VARDATA(value_datum), sqlite3_value_blob(val), value_byte_size_blob_or_utf8);
			SET_VARSIZE(value_datum, value_byte_size_blob_or_utf8 + VARHDRSZ);
			return (struct NullableDatum) {PointerGetDatum((const void *)value_datum), false};
		}
		else
			ereport(ERROR, (errcode(ERRCODE_FDW_INVALID_DATA_TYPE),
				errmsg("This PostGIS data type is supported by SpatiaLite, but FDW compiled without GIS data support"),
				errhint("Data type: \"%s\" in column \"%.*s\"", pg_dataTypeName, (int)sizeof(pgColND.data), pgColND.data)));
```
Similarly, for INSERT query, we can modify `sqlite_bind_sql_var` function.
```
	Oid			atttypid = att->atttypid;
	int32		atttypmod = att->atttypmod;

	/*
	 * If a domain has been declared as bytea, it can support PostGIS data type
	 */
	atttypid = getBaseTypeAndTypmod(atttypid, &atttypmod);

	if (atttypid == BYTEAOID)
	{
		blobOutput b = sqlite_datum_to_blob(value);
		ret = sqlite3_bind_blob(stmt, attnum, b.dat, b.len, SQLITE_TRANSIENT);
	}
	else
		ereport(ERROR, (errcode(ERRCODE_FDW_INVALID_DATA_TYPE),
				errmsg("This PostGIS data type is supported by SpatiaLite, but FDW compiled without GIS data support"),
				errhint("Data type: \"%s\" in column \"%.*s\"", pg_dataTypeName, (int)sizeof(pgColND.data), pgColND.data)));
```
After above modification, the result is consistent between declaring or not declaring domain.  
Testcase 31: `gm` and `gg` column can be bound successfully. Error occurred because of `r` column with raster type, same as in postgis.sql file.
```
--Testcase 31: ERR - geometry
INSERT INTO "types_PostGIS" ( "i", gm, gg, r, t ) VALUES (1, decode('0101000020e6100000fd5aa846f9733e406c054d4bacd74d40', 'hex'),  decode('0101000020e6100000fd5aa846f9733e406c054d4bacd74d40', 'hex'),  decode('1223456890', 'hex'), '{"genus": "Rhododendron", "taxon": "Rhododendron ledebourii", "natural": "shrub", "genus:ru": "Рододендрон", "taxon:ru": "Рододендрон Ледебура", "source:taxon": "board"}');
ERROR:  This data type is PostGIS specific and have not any SpatiaLite value
HINT:  Data type: "public.raster" in column "r"
```
Testcase 37 can return data without any problem, consistent with test case 44.
```
--Testcase 37: ERR - no GIS data support
SELECT "i", gm, gg, t FROM "types_PostGIS";
 i |                                                             gm                                                             |                                                             gg                                                             |                                                                                     t                                                                                     
---+----------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 1 | \x0001e6100000bf72ce99fe763e40ed4960730ed84d40bf72ce99fe763e40ed4960730ed84d407c01000000bf72ce99fe763e40ed4960730ed84d40fe | \x0001e6100000bf72ce99fe763e40ed4960730ed84d40bf72ce99fe763e40ed4960730ed84d407c01000000bf72ce99fe763e40ed4960730ed84d40fe | {"genus": "Rhododendron", "taxon": "Rhododendron ledebourii", "natural": "shrub", "genus:ru": "Рододендрон", "taxon:ru": "Рододендрон Ледебура", "source:taxon": "board"}
(1 row)
```
Please update it.